src/coreclr/vm/gcheaputilities.h

```diff
@@ -26,6 +26,14 @@ GVAL_DECL(gc_alloc_context, g_global_alloc_context);
 }
 #endif // !DACCESS_COMPILE
 
+// on MP systems, each thread has its own allocation chunk so we can avoid
+// lock prefixes and expensive MP cache snooping stuff
+#ifdef _MSC_VER
+EXTERN_C __declspec(selectany) __declspec(thread) gc_alloc_context t_thread_alloc_context;
```

Comment:
IIRC, `__declspec(selectany` was a workaround for some thread statics code quality problem. I am wondering whether it is still needed.

Reply:
I checked and it's still needed sadly (that's the only reason I added it here).